File: erettsegi/kozep/2023_may/java/Kitalalo.java
import module java.base;

public class Kitalalo {

    public static void main(String[] args) throws Exception {
        var szavak = Files.readString(Path.of("szavak.txt")).trim().replace("\"", "").split(", ");
        var random = new Random();
        var kivalasztottSzo = szavak[random.nextInt(szavak.length)];
        var probakSzama = 1;

        while(true) {
            var tipp = System.console().readLine("Kérem a tippet: ");

            if(tipp.equals("stop")) {
                break;
            }else{
                System.out.print("Az eredmény: ");

                var eltalaltKarakterek = 0;
                for(var i = 0; i < 6; ++i) {
                    var eredetiKarakter = kivalasztottSzo.charAt(i);
                    var eltalaltaE = eredetiKarakter == tipp.charAt(i);

                    System.out.print(eltalaltaE ? eredetiKarakter : '.');

                    if(eltalaltaE) {
                        ++eltalaltKarakterek;
                    }
                }

                System.out.println();

                if(eltalaltKarakterek == 6) {
                    System.out.println("\n" + probakSzama + " tippeléssel sikerült kitalálni");
                    break;
                }else{
                    ++probakSzama;
                }
            }
        }
    }
}

File: erettsegi/kozep/2023_may/c#/Kitalalo.cs
var szavak = File.ReadAllText("szavak.txt").Trim().Replace("\"", "").Split(", ");
var random = new Random();
var kivalasztottSzo = szavak[random.Next(szavak.Length)];
var probakSzama = 1;

while(true) {
    Console.Write("Kérem a tippet: ");
    var tipp = Console.ReadLine()!;

    if(tipp == "stop") {
        break;
    }else{
        Console.Write("Az eredmény: ");

        var eltalaltKarakterek = 0;
        for(var i = 0; i < 6; ++i) {
            var eredetiKarakter = kivalasztottSzo[i];
            var eltalaltaE = eredetiKarakter == tipp[i];

            Console.Write(eltalaltaE ? eredetiKarakter : '.');

            if(eltalaltaE) {
                ++eltalaltKarakterek;
            }
        }

        Console.WriteLine();

        if(eltalaltKarakterek == 6) {
            Console.WriteLine("\n" + probakSzama + " tippeléssel sikerült kitalálni");
            break;
        }else{
            ++probakSzama;
        }
    }
}
